forage-android/src/main/java/com/joinforage/forage/android/LDManager.kt

```diff
@@ -72,17 +85,54 @@ internal object LDManager {
         }
         val contextKind = ContextKind.of(LDContextKind.SERVICE)
         val context = LDContext.create(contextKind, LDContexts.ANDROID_CONTEXT)
-        val client = LDClient.init(app, ldConfig, context, 0)
+        client = LDClient.init(app, ldConfig, context, 0)
+
+        internalLogger = logger
+    }
+
+    // vaultType is instantiated lazily and is a singleton. Once we set the vault type once, we don't
+    // want to overwrite it! We must take in the application as a parameter, which means that a
+    // ForagePINEditText must be rendered before any of the ForageSDKApi functions are called.
+    internal fun getVaultProvider(): VaultType {
+        if (vaultType != null) {
+            return vaultType as VaultType
+        }
         // default to 100% VGS usage in case LD flag retrieval fails
-        val vaultPercent = client.doubleVariation(LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG, 0.0)
+        val vaultPercent =
+            client?.doubleVariation(LDFlags.VAULT_PRIMARY_TRAFFIC_PERCENTAGE_FLAG, 0.0)
         internalLogger.i("[LaunchDarkly] Vault percent of $vaultPercent return from LD")
         val randomNum = Math.random() * 100
-        vaultType = if (randomNum < vaultPercent) {
+
+        vaultType = if (randomNum < (vaultPercent ?: 0.0)) {
             VaultType.BT_VAULT_TYPE
         } else {
             VaultType.VGS_VAULT_TYPE
         }
         internalLogger.i("[LaunchDarkly] Vault type set to $vaultType")
         return vaultType as VaultType
     }
+
+    internal fun getPollingIntervalsFeatureFlag(): IntArray {
+        if (pollingIntervals != null) {
```

Comment:
Oh i thought this was the pattern for all feature flags in general. I can just return the latest polling intervals every time instead of storing the value in this singleton pattern